File: scripts/calculate-signatures.py
#!/usr/bin/env python
import sys
sys.path.append('test')
from utils.utilities import str_to_felt
from utils.Signer import Signer
from starkware.cairo.common.hash_state import compute_hash_on_elements

owner = Signer(1)
guardian = Signer(2)
guardian_backup = Signer(3)

new_owner = Signer(4)
new_guardian = Signer(5)
new_guardian_backup = Signer(6)

wrong_owner = Signer(7)
wrong_guardian = Signer(8)

ESCAPE_SECURITY_PERIOD = 24*7*60*60

VERSION = str_to_felt('0.2.4')
NAME = str_to_felt('ArgentAccount')

IACCOUNT_ID = 0xa66bd575
IACCOUNT_ID_OLD = 0x3943f10f

ESCAPE_TYPE_GUARDIAN = 1
ESCAPE_TYPE_OWNER = 2


def calculate_sig_account():
    hash = 1283225199545181604979924458180358646374088657288769423115053097913173815464
    invalid_hash = 1283225199545181604979924458180358646374088657288769423115053097913173811111

    owner_r, owner_s = owner.sign(hash)
    guardian_r, guardian_s = guardian.sign(hash)
    guardian_backup_r, guardian_backup_s = guardian_backup.sign(hash)
    wrong_owner_r, wrong_owner_s = wrong_owner.sign(hash)
    wrong_guardian_r, wrong_guardian_s = wrong_guardian.sign(hash)

    print(f"""

    const message_hash: felt252 = 0x{hash:x};

    const owner_pubkey: felt252 = 0x{owner.public_key:x};
    const owner_r: felt252 = 0x{owner_r:x};
    const owner_s: felt252 = 0x{owner_s:x};

    const guardian_pubkey: felt252 = 0x{guardian.public_key:x};
    const guardian_r: felt252 = 0x{guardian_r:x};
    const guardian_s: felt252 = 0x{guardian_s:x};

    const guardian_backup_pubkey: felt252 = 0x{guardian_backup.public_key:x};
    const guardian_backup_r: felt252 = 0x{guardian_backup_r:x};
    const guardian_backup_s: felt252 = 0x{guardian_backup_s:x};

    const wrong_owner_pubkey: felt252 = 0x{wrong_owner.public_key:x};
    const wrong_owner_r: felt252 = 0x{wrong_owner_r:x};
    const wrong_owner_s: felt252 = 0x{wrong_owner_s:x};

    const wrong_guardian_pubkey: felt252 = 0x{wrong_guardian.public_key:x};
    const wrong_guardian_r: felt252 = 0x{wrong_guardian_r:x};
    const wrong_guardian_s: felt252 = 0x{wrong_guardian_s:x};

    """)


def calculate_sig_change_owner():
    # message_hash = pedersen(0, (change_owner selector, chainid, contract address, old_owner))
    change_owner_selector = 658036363289841962501247229249022783727527757834043681434485756469236076608
    chain_id = 0
    contract_address = 1
    old_owner = 0x1ef15c18599971b7beced415a40f0c7deacfd9b0d1819e03d723d8bc943cfca
    
    message_hash = compute_hash_on_elements([change_owner_selector, chain_id, contract_address, old_owner])

    new_owner_r, new_owner_s = new_owner.sign(message_hash)

    print(f"""

    const new_owner_pubkey: felt252 = 0x{new_owner.public_key:x};
    const new_owner_r: felt252 = 0x{new_owner_r:x};
    const new_owner_s: felt252 = 0x{new_owner_s:x};
    """)


calculate_sig_change_owner()

File: scripts/calculate-signatures.ts
import { hash, shortString, num } from "starknet";

import { KeyPair } from "../tests/lib/signers";

const owner = new KeyPair(1n);
const guardian = new KeyPair(2n);
const guardian_backup = new KeyPair(3n);

const new_owner = new KeyPair(4n);
const new_guardian = new KeyPair(5n);
const new_guardian_backup = new KeyPair(6n);

const wrong_owner = new KeyPair(7n);
const wrong_guardian = new KeyPair(8n);

const ESCAPE_SECURITY_PERIOD = 24 * 7 * 60 * 60;

const VERSION = shortString.encodeShortString("0.2.4");
const NAME = shortString.encodeShortString("ArgentAccount");

const IACCOUNT_ID = 0xa66bd575;
const IACCOUNT_ID_OLD = 0x3943f10f;

const ESCAPE_TYPE_GUARDIAN = 1;
const ESCAPE_TYPE_OWNER = 2;

function calculate_sig_account() {
  const hash = "0x2d6479c0758efbb5aa07d35ed5454d728637fceab7ba544d3ea95403a5630a8";
  const invalid_hash = "0x02d6479c0758efbb5aa07d35ed5454d728637fceab7ba544d3ea95403a561fa7";

  const [owner_r, owner_s] = owner.signHash(hash);
  const [guardian_r, guardian_s] = guardian.signHash(hash);
  const [guardian_backup_r, guardian_backup_s] = guardian_backup.signHash(hash);
  const [wrong_owner_r, wrong_owner_s] = wrong_owner.signHash(hash);
  const [wrong_guardian_r, wrong_guardian_s] = wrong_guardian.signHash(hash);

  console.log(`
    const message_hash: felt252 = ${num.toHex(hash)};
    const invalid_hash: felt252 = ${num.toHex(invalid_hash)};

    const owner_pubkey: felt252 = ${num.toHex(owner.publicKey)};
    const owner_r: felt252 = ${num.toHex(owner_r)};
    const owner_s: felt252 = ${num.toHex(owner_s)};

    const guardian_pubkey: felt252 = ${num.toHex(guardian.publicKey)};
    const guardian_r: felt252 = ${num.toHex(guardian_r)};
    const guardian_s: felt252 = ${num.toHex(guardian_s)};

    const guardian_backup_pubkey: felt252 = ${num.toHex(guardian_backup.publicKey)};
    const guardian_backup_r: felt252 = ${num.toHex(guardian_backup_r)};
    const guardian_backup_s: felt252 = ${num.toHex(guardian_backup_s)};

    const wrong_owner_pubkey: felt252 = ${num.toHex(wrong_owner.publicKey)};
    const wrong_owner_r: felt252 = ${num.toHex(wrong_owner_r)};
    const wrong_owner_s: felt252 = ${num.toHex(wrong_owner_s)};

    const wrong_guardian_pubkey: felt252 = ${num.toHex(wrong_guardian.publicKey)};
    const wrong_guardian_r: felt252 = ${num.toHex(wrong_guardian_r)};
    const wrong_guardian_s: felt252 = ${num.toHex(wrong_guardian_s)};
`);
}

function calculate_sig_change_owner() {
  // message_hash = pedersen(0, (change_owner selector, chainid, contract address, old_owner))
  const change_owner_selector = hash.getSelector("change_owner");
  const chain_id = 0;
  const contract_address = 1n;
  const old_owner = "0x1ef15c18599971b7beced415a40f0c7deacfd9b0d1819e03d723d8bc943cfca";

  const message_hash = hash.computeHashOnElements([change_owner_selector, chain_id, contract_address, old_owner]);

  const [new_owner_r, new_owner_s] = new_owner.signHash(message_hash);

  console.log(`

    const new_owner_pubkey: felt252 = ${num.toHex(new_owner.publicKey)};
    const new_owner_r: felt252 = ${num.toHex(new_owner_r)};
    const new_owner_s: felt252 = ${num.toHex(new_owner_s)}; 
    `);
}

calculate_sig_account();
